sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -254,6 +312,12 @@ object TableOutputResolver {
       !Cast.canUpCast(cast.child.dataType, cast.dataType)
   }
 
+  private def isCompatible(tableAttr: Attribute, queryExpr: NamedExpression): Boolean = {
```

Comment:
Moved the existing logic into a separate method as I added a nested if/else block and it became hard to read.